src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/ClientTrackingWebFilter.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config
+
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.trace.Span
+import org.slf4j.LoggerFactory
+import org.springframework.http.HttpHeaders
+import org.springframework.stereotype.Component
+import org.springframework.web.server.ServerWebExchange
+import org.springframework.web.server.WebFilter
+import org.springframework.web.server.WebFilterChain
+import reactor.core.publisher.Mono
+import java.text.ParseException
+
+@Component
+class ClientTrackingWebFilter : WebFilter {
+  private val bearer = "Bearer "
+  private val clientIdKey = AttributeKey.stringKey("clientId")
+  private val usernameKey = AttributeKey.stringKey("username")
+
+  override fun filter(exchange: ServerWebExchange, chain: WebFilterChain): Mono<Void> {
+    val token = exchange.request.headers.getFirst(HttpHeaders.AUTHORIZATION)
+    if (token?.startsWith(bearer, ignoreCase = true) == true) {
+      try {
+        val jwtBody = getClaimsFromJWT(token)
+        val currentSpan = Span.current()
+        val user = jwtBody.getClaim("user_name")
+        user?.run { currentSpan.setAttribute(usernameKey, user.toString()) }
+        currentSpan.setAttribute(clientIdKey, jwtBody.getClaim("client_id").toString())
```

Comment:
note using opentelemetry to set attributes instead of app insights sdk.  This is as per the latest documentation in https://docs.microsoft.com/en-us/azure/azure-monitor/app/java-in-process-agent#add-a-custom-dimension

Reply:
can't get the test working at present since requires `Span` to be set, hence why test was removed.